src/lib/statistics/cardinality_estimation_cache.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "join_graph_statistics_cache.hpp"
+
+namespace opossum {
+
+class CardinalityEstimationCache {
+ public:
+  /**
+   * Join/Predicate Ordering rules can enable this cache by calling `join_graph_statistics_cache.emplace()`. The docs of
+   * JoinGraphStatisticsCache explain what this cache does.
+   */
+  std::optional<JoinGraphStatisticsCache> join_graph_statistics_cache;
+
+  /**
+   * Optimization Rules that !!!do not manipulate subplans after they have build them!!!! can enable
```

Comment:
If it's really that important, use caps. Computers should burst into flames when people press `!` twice. ;)